### PR TITLE
Nuzhy/live market block with live market cards

### PR DESCRIPTION
**Pull request by Nuzhy-Deriv**
live-market block with cards
image

### CODE REVIEW

File: libs/blocks/src/lib/live-market/index.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import { StandaloneChevronRightRegularIcon } from '@deriv/quill-icons';
+import clsx from 'clsx';
+import { ReactNode } from 'react';
+
+export interface LiveMarketProps {
+  isHeaderSection?: boolean;
+  header?: string;
+  description?: string;
+  className?: string;
+  tabHeader?: ReactNode;
```
Comment (prince-deriv) [inline libs/blocks/src/lib/live-market/index.tsx:12]:
```suggestion
  tabHeader?: () => ReactNode;
```

File: libs/blocks/src/lib/live-market/index.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        'bg-background-primary-container',
+        className,
+      )}
+    >
+      <FluidContainer className="flex flex-col items-center justify-center gap-gap-2xl">
+        {isHeaderSection && (
+          <div className="flex flex-col items-center gap-gap-lg">
+            {header && <Heading.H2>{header}</Heading.H2>}
+            {description && <Text size="md">{description}</Text>}
+            {tabHeader}
```
Comment (prince-deriv) [inline libs/blocks/src/lib/live-market/index.tsx:44]:
```suggestion
           <tabHeader />
```

**Review by prince-deriv — DISMISSED** [review]

File: libs/blocks/src/lib/features/content-slider/index.tsx [hunk 3]
```diff
@@ -1,23 +1,27 @@
...
+  CardSlider,
+  CardSliderProps,
+  CardVariantType,
+} from '@deriv-com/components';
 
-export interface ContentSliderProps {
+export interface ContentSliderProps<T extends CardVariantType> {
   title?: string;
   description?: string;
   cta?: ReactNode;
```
Comment (prince-deriv) [inline libs/blocks/src/lib/features/content-slider/index.tsx:13]:
```suggestion
  cta?:  () => ReactNode;
```

File: libs/blocks/src/lib/features/content-slider/index.tsx [hunk 4]
```diff
@@ -35,7 +39,11 @@ const ContentSlider = ({
             </Text>
           )}
         </div>
-        <CardSlider {...cardSliderProps} />
+        {
+          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+          // @ts-ignore
+          <CardSlider {...cardSliderProps} />
+        }
         {cta}
```
Comment (prince-deriv) [inline libs/blocks/src/lib/features/content-slider/index.tsx:47]:
```suggestion
        {cta && <cta />}
```

File: libs/components/src/lib/card/live-market/index.tsx [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            <StandaloneArrowDownRegularIcon
+              iconSize="sm"
+              fill={colorVariant[status]}
+            />
+          ) : (
+            <StandaloneArrowRightRegularIcon
+              iconSize="sm"
+              fill={colorVariant[status]}
+            />
+          )}
```
Comment (prince-deriv) [inline libs/components/src/lib/card/live-market/index.tsx:68]: What is the difference here? No need to add a condition if we are displaying thesame for all of them. 
```suggestion
       
            <StandaloneArrowRightRegularIcon
              iconSize="sm"
              fill={colorVariant[status]}
            />
         
```

Comment (Nuzhy-Deriv) [reply libs/components/src/lib/card/live-market/index.tsx:68]: Different icon based on the status. currently I have just used the sample icons. will update once the charticons are ready in quill-icons.

Comment (prince-deriv) [reply libs/components/src/lib/card/live-market/index.tsx]: Those icons can be added as an object 

```
 const icons = {
  up: <StandaloneArrowDownRegularIcon
              iconSize="sm"
              fill={colorVariant[status]}
            />,
    down: <StandaloneArrowDownRegularIcon
              iconSize="sm"
              fill={colorVariant[status]}
            />
            }

**Review by prince-deriv — DISMISSED** [review]

File: libs/blocks/src/lib/live-market/index.tsx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+  return (
+    <Section
+      className={clsx(
+        'py-general-4xl',
+        'bg-background-primary-container',
+        className,
+      )}
+    >
+      <FluidContainer className="flex flex-col items-center justify-center gap-gap-xl lg:gap-gap-2xl">
+        {isHeaderSection && (
```
Comment (prince-deriv) [inline libs/blocks/src/lib/live-market/index.tsx:45]:
```suggestion
        {(header || description || TabHeader) &&  (
```

File: libs/components/src/lib/card-slider/index.tsx [hunk 7]
```diff
@@ -38,31 +29,34 @@ const defaultSwiperProps: SwiperProps = {
...
+export const CardSlider = <T extends CardVariantType>({
+  variant,
   cards = [],
-  renderCard,
+  className,
   slideClasses,
   swiperData,
-}: CardSliderProps) => {
+}: CardSliderProps<T>) => {
   const swiperProps = Object.assign(defaultSwiperProps, swiperData);
```
Comment (prince-deriv) [inline libs/components/src/lib/card-slider/index.tsx:39]:
```suggestion
  const {pagination, slidesPerView,spaceBetween,breakpoints} = Object.assign(defaultSwiperProps, swiperData);
```

File: libs/components/src/lib/card-slider/index.tsx [hunk 8]
```diff
@@ -38,31 +29,34 @@ const defaultSwiperProps: SwiperProps = {
...
-        className={qtMerge(
-          swiperProps.pagination && '!pb-general-3xl',
-          className,
-        )}
+        className={qtMerge(className)}
         modules={[Pagination]}
         pagination={swiperProps.pagination}
         slidesPerView={swiperProps.slidesPerView}
         spaceBetween={swiperProps.spaceBetween}
+        breakpoints={swiperProps.breakpoints}
```
Comment (prince-deriv) [inline libs/components/src/lib/card-slider/index.tsx:50]:
```suggestion
        pagination={pagination}
        slidesPerView={slidesPerView}
        spaceBetween={spaceBetween}
        breakpoints={breakpoints}
```

File: libs/components/src/lib/card/live-market/live-price.tsx [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+      <div className="flex flex-row items-center gap-[2px]">
+        <Text size="md" className={clsx('pt-[6px]', colorVariant[status])}>
+          {price?.substring(0, 4)}
+        </Text>
+        <Text size="xl" className={clsx(colorVariant[status])}>
+          {price?.substring(4, 6)}
+        </Text>
+        <Text size="md" className={clsx('pb-[6px]', colorVariant[status])}>
+          {price?.substring(6)}
+        </Text>
```
Comment (prince-deriv) [inline libs/components/src/lib/card/live-market/live-price.tsx:30]: Later we need to have a way to add configuration on this.  enlargement of the numbers will vary on the instrument type.

**Review by prince-deriv — APPROVED** [review]

File: libs/components/src/lib/card/live-market/index.tsx [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+  className,
+  instrumentIcon,
+  instrument,
+  changePercentage,
+  status,
+  bidPrice,
+  askPrice,
+  spread,
+}) => {
+  const colorVariant = {
```
Comment (mohsen-deriv) [inline libs/components/src/lib/card/live-market/index.tsx:24]: please move these out of the component, these are not connected to any props or states, thanks 🙏

File: libs/components/src/lib/card/live-market/index.tsx [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+    up: 'fill-solid-emerald-900 text-solid-emerald-900',
+    down: 'fill-solid-cherry-700 text-solid-cherry-700',
+    remain: 'fill-opacity-black-400 text-opacity-black-400',
+    closed: 'fill-opacity-black-300 text-opacity-black-300',
+  };
+
+  const ChartIcons = {
```
Comment (mohsen-deriv) [inline libs/components/src/lib/card/live-market/index.tsx:31]: please move these out to outside of the component, thanks 🙏

File: libs/components/src/lib/card/live-market/index.tsx [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+        'rounded-xl',
+        'h-[262px] w-[286px]',
+        'bg-background-primary-container',
+        'p-general-xl',
+        'gap-gap-xl',
+        className,
+        status === 'closed' ? 'opacity-700' : '',
+      )}
+    >
+      <div className="flex h-[52px] flex-row gap-gap-sm">
```
Comment (mohsen-deriv) [inline libs/components/src/lib/card/live-market/index.tsx:65]: can you please break down this big component to smaller one? 
thanks thanks 🙏

**Pull request merged**